flink-kubernetes-standalone/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/decorators/CmdStandaloneJobManagerDecorator.java

```diff
@@ -80,9 +80,14 @@ private List<String> getApplicationClusterArgs() {
         }
 
         Boolean allowNonRestoredState = kubernetesJobManagerParameters.getAllowNonRestoredState();
-        if (allowNonRestoredState != null) {
+        if (allowNonRestoredState != null && allowNonRestoredState) {
             args.add("--allowNonRestoredState");
-            args.add(allowNonRestoredState.toString());
+        }
+
+        String savepointPath = kubernetesJobManagerParameters.getSavepointPath();
+        if (savepointPath != null) {
+            args.add("--fromSavepoint");
+            args.add(savepointPath);
```

Comment:
What I meant with my other question is actually why we need this part here :) 
Seems like before this change the savepoint restore logic worked correctly so the config is probably enough.

Wouldn't be enough to simply fix the allowNonRestoredState logic to not put true/false there?

Reply:
As we discussed on Slack, before this fix. Both `initialSavepoint` and savepoint upgrade logic worked correctly so there must not really be a need for the `--fromSavepoint` argument

Reply:
Yep, I think you're right: it seems like just fixing the `allowNonRestoredState` flag would be enough. 

However, I think it's logical to provide the two options together via the command-line. They always come together (check `SavepointConfigOptions`, `SavepointRestoreSettings`, etc.) and I was really surprised when I didn't see the `fromSavepoint` argument when debugging. 

But if you insist on making the smallest change in this PR then I'll revert the `fromSavepoint` part. Let me know!

Reply:
I don't insist I think it's good like this for more visibility, I just wanted to make sure I fully understand the reason behind this particular change. :)

Reply:
actually I think it might be enough to remove both `fromSavepoint` and `allowNonRestoredState` command line arguments and rely purely on the config. If that works I would actually lean toward that. It's much simpler and more bulletproof to just use the conf

Reply:
I thought about it too, but based on these lines:

- https://github.com/apache/flink/blob/master/flink-container/src/main/java/org/apache/flink/container/entrypoint/StandaloneApplicationClusterEntryPoint.java#L95
- https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java#L166-L168

it looks like if the flag is not present on the command-line, Flink will override the value from the config file to `false`.

Reply:
You are right, let’s keep it as is